### DISCUSSION TITLE
How to navigate a comments programmatically?

### DISCUSSION THREAD

**Discussion by KrabsWong** [microsoft/vscode-discussions#184] [Extension Development QnA] [answered]
Click comment item in the Comments panel will open the target file and navigate to the target comment, is there any way to do it programmatically?

The scenario from my side is listed below:

1. Generated a custom panel implemented by TreeView provider to show up comments and other information, I wanna to navigate the target file and line by click the treeview item.
2. In the file editor, click next/previous icon to navigate the next/previous comment.

**Comment by alefragnani** [comment, marked as answer]
Hi @KrabsWong ,

To open a file and navigate to some specific position, you must open a file, show it and _reveal_ the position (line - character). Something like:

```ts
    const document = await vscode.workspace.openTextDocument(fileUri);
    await vscode.window.showTextDocument(document);

    const position = new Selection(line, column, line, column);
    window.activeTextEditor.selection = newPosition;
    window.activeTextEditor.revealRange(newPosition, reviewType);
```

About the comment itself, I didn't play with it that much, but it seems the `CommentThread` interface has a `collapsibleState`, which defines if the thread must be displayed collapsed/expanded when opened the document.

BTW, I would suggest you to take a look at the CodeTour extension (https://github.com/microsoft/codetour), which I remember to be a good source of knowledge about the Comments API, as it uses it at its core.

Hope this helps

**Reply by KrabsWong** [reply]
Yes, I am trying to use the selection and revealRange to reach my goal.

You are always very prompt with feedback, thank you sooooo much
